File: développement/backend/stage/src/main/java/stage/stage/entity/Stagaire.java
package stage.stage.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import java.io.Serializable;




@Entity
@DynamicUpdate
@DynamicInsert
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Stagaire implements Serializable {
    private static final Long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer stagaireId;

    private Integer stagaireCIN;
    private String stagaireFirstName;
    private String stagaireLastName;
    private String email;
    private String password;
    private String university;
    private String specialite;

    private String phoneNumber;
    private String etat ;

    private String role;

    private String typeStage;
    private String date_Debut_Stage;
    private String date_Fin_Stage;

    private String date_demande_stage;
}

File: développement/frontend/firstP/src/app/Modals/stagaire.ts
export class stagaire {
    stagaireId: number;
    stagaireCIN: number;
    stagaireFirstName: string;
    stagaireLastName: string;
    email: string;
    password: string;
    university: string;
    specialite: string;
    phoneNumber: number;
    etat: string;
    role: string;
    typeStage: string;
    date_Debut_Stage: any;
    date_Fin_Stage: any;
    date_demande_stage: any;

    constructor() {
        this.stagaireId =0
        this.stagaireCIN = 0;
        this.stagaireFirstName = "";
        this.stagaireLastName = "";
        this.email = "";
        this.password = "";
        this.university = "";
        this.specialite = "";
        this.phoneNumber = 0;
        this.etat = "ENVOYER";
        this.role = "Stagaire";
        this.typeStage = "";
        this.date_Debut_Stage = null;
        this.date_Fin_Stage = null;
        this.date_demande_stage = null;
    }
}
